lib/src/oauth2.dart

```diff
@@ -120,18 +122,26 @@ Future<T> withClient<T>(SystemCache cache, Future<T> Function(Client) fn) {
 ///
 /// If saved credentials are available, those are used; otherwise, the user is
 /// prompted to authorize the pub client.
-Future<Client> _getClient(SystemCache cache) async {
-  var credentials = _loadCredentials(cache);
-  if (credentials == null) return await _authorize();
-
-  var client = Client(credentials,
-      identifier: _identifier,
-      secret: _secret,
-      // Google's OAuth2 API doesn't support basic auth.
-      basicAuth: false,
-      httpClient: httpClient);
-  _saveCredentials(cache, client.credentials);
-  return client;
+Future<Client> _getClient(SystemCache cache, Uri requestServer) async {
+  var isThirdPartyServer =
+      !runningFromTest && requestServer.host != 'https://pub.dartlang.org';
```

Comment:
@jonasfj won't this break pre-release testing? (and similarly in the `getToken` method)

Reply:
Actually it won't, or at least, it is still the same behaviour as before.
Checking the `requestServer` passed in, you will see that it is set through the `server` field in the `ListCommand`
```
  /// The URL of the server to which to upload the package.
  Uri get server {
    // An explicit argument takes precedence.
    if (argResults.wasParsed('server')) {
      return Uri.parse(argResults['server']);
    }

    // Otherwise, use the one specified in the pubspec.
    if (entrypoint.root.pubspec.publishTo != null) {
      return Uri.parse(entrypoint.root.pubspec.publishTo);
    }

    // Otherwise, use the default.
    return Uri.parse(cache.sources.hosted.defaultUrl);
  }
```
The last line (default) points to
```
    return _defaultUrl ??= _pubHostedUrlConfig() ?? 'https://pub.dartlang.org';

```
in hosted.dart => so in the end it is the same compare (towards https://pub.dartlang.org).